cardano-api/src/Cardano/Api/Modes.hs

```diff
@@ -154,6 +153,58 @@ data EraInMode era mode where
 
 deriving instance Show (EraInMode era mode)
 
+deriving instance Eq (EraInMode era mode)
+
+instance FromJSON (EraInMode ByronEra ByronMode) where
+  parseJSON "ByronEraInByronMode" = pure ByronEraInByronMode
+  parseJSON invalid =
+      prependFailure "parsing 'EraInMode ByronEra ByronMode failed, "
```

Comment:
Can't we abstract this to something like: `invalidJSONFailure :: String -> String -> Parser a`?

Reply:
Good point! Are the changes okay?